spec/jobs/user_stat_collection_job_spec.rb

```diff
@@ -13,10 +13,23 @@
 
   let(:account) { create(:account_with_public_schema) }
 
-  describe '#reenqueue' do
-    it 'Enques an TenantMaintenanceJob after perform' do
-      switch!(account)
-      expect { UserStatCollectionJob.perform_now }.to have_enqueued_job(UserStatCollectionJob)
+  describe '#perform' do
+    context 'when UserStat records exist' do
+      before do
+        FactoryBot.create(:user_stat) # Ensure a UserStat record exists
+      end
+
+      it 'enqueues UserStatCollectionJob after perform' do
+        switch!(account)
```

Comment:
My only hesitation is that sometimes when you're switching to a different tenant in a test, it can leaky into other tests making them flaky.  I added this in another test:
https://github.com/samvera/hyku/blob/acaaf6a5af1127377cb687773f3c308a9b2a6779/spec/services/create_account_spec.rb#L11